### ISSUE TITLE
filter py3 & finding anaconda python

### ISSUE THREAD

**Issue by ahed87** [Nodd/spyder_line_profiler#28]
Hi,
this really looks like it should be a standard plugin to spyder.
it's an awesome functionality to have in spyder!!
Very easy to use once it's up and running.

One possible further step could be non-intrusive marking of the @profile.
Maybe it would be possible to make a mark similar to "breakpoint" in debugging?

Here is what I had to do to get it running on my computer.
I had 2 issues (both with memory_profiler, the line_profiler worked out of the box):

First of all I run anaconda 2.3.0 py34 64-bit on win8.1
I installed line_profiler & psutil with conda. The memory_profiler was installed with pip, since conda did not have a version in the repository (also the ones found on binstar seemed to be old).

As for getting into spyder I just copied the 4 relevant files in this repository to the spyder plug-in location as per the instructions.

#1 (this library). in memoryprofilergui changed line:
from: stuff = filter(None, l.split(' '))
to: stuff = list(filter(None, l.split(' ')))
I guess this is a py2 vs py3 thing, and I took the easy way out with making a list to go back to py2 behaviour, I did not test tuple but I guess that works as well.

#2 (more spyder itself) - it did not find the right python.exe.
The line_profiler worked out of the box.
After a while I tracked it down to that for the line_profiler it started the program "kernprof",
this one I only have one version of, and that's in the anaconda installation.
The memory_profiler started the program "python", which it finds with 
spyderlib.utils.programs.find_program('python').

I know that I have a pretty scattered computer with regards to python installs, but it normally works out OK.
I have 3 different base pythons installed where anaconda and a regular python are 2 of them.
There are also a couple of virtual envs from the "small python" base.

Anyways when spyderlib finds the "python" progam it finds the path 'C:\Python34\python.exe'
If I put in IPython I get 'C:\\Data\\Anaconda3\\Scripts\\ipython.exe',
in other words - although spyder is installed via anaconda, it still does not recognize this...

When I look into my os.environ['PATH'] the non-anaconda paths are before the anaconda paths in a print-out.
So I guess it finds the non-anaconda first and is satisfied with that answer.

Maybe one can use the 'SPYDER_PARENT_DIR' environ_var to select the right python install.
For myself I just put in the hard path, i.e. line no 264 executable = "c:\..\Anaconda3\python.exe" to make it work on my computer.

Once more, thanks for a great integration into spyder!!

**Comment by pedvide** [comment]
I confirm that with 
             platform : win-64
        conda version : 3.18.9
  conda-build version : 1.18.2
       python version : 3.5.1.final.0
     requests version : 2.9.0
installing line_profiler with conda install, and memory_profiler with pip install
and copying the files to the folder Anaconda3\Lib\site-packages\spyderplugins
line_profiler works well and memory_profiler works when doing the change
in memoryprofilergui changed line:
from: stuff = filter(None, l.split(' '))
to: stuff = list(filter(None, l.split(' ')))

**Comment by divenex** [comment]
I started a pull request on this [here](https://github.com/Nodd/spyder_line_profiler/pull/30).

**Comment by Nodd** [comment]
Thanks for the reports, I merged the PR made by @divenex to fix the python3 compatibility.

The `@profile` decorator is needed by the underlying `line_profiler` library, I don't think it can be avoided. See https://github.com/rkern/line_profiler/issues/51 for having less pain with it. I just put these lines at the top of my files to be able to elave the decorator when run normally:

``` python
try:
    profile
except NameError:
    profile = lambda x: x
```

About multiple python installs, it is true that right now it takes the first executable found, it's not very clean. It should use the value configured in spyder, I suppose?

Also please note that plugin support has been improved in spyder 3.0 (currently in beta phase). The plugin compatible for the new version is located at https://github.com/spyder-ide/spyder.line_profiler, so any new improvements should be discussed over there.
